ui/src/views/Documentation/index.jsx

```diff
@@ -106,63 +182,220 @@ export default class Documentation extends Component {
     return this.findChildFromRootNode(rootNode) || rootNode;
   }
 
+  buildSchemaId(schemaId) {
+    if (schemaId.startsWith('/')) {
+      if (
+        process.env.TASKCLUSTER_ROOT_URL &&
+        process.env.TASKCLUSTER_ROOT_URL !== 'https://taskcluster.net'
+      ) {
+        return process.env.TASKCLUSTER_ROOT_URL + schemaId;
+      }
+
+      return `https://schemas.taskcluster.net/${schemaId.replace(
+        /^\/schemas\//,
+        ''
+      )}`;
+    }
+
+    return schemaId;
+  }
+
+  sanitizeSchema(schema) {
+    if (schema.$id) {
+      return {
+        ...schema,
+        $id: this.buildSchemaId(schema.$id),
+      };
+    }
+
+    if (schema.id) {
+      return {
+        ...schema,
+        id: this.buildSchemaId(schema.id),
+      };
+    }
+
+    return schema;
+  }
+
+  // Returns a mapping between the HTML element and the desired component
+  components() {
+    return {
+      a: this.anchorFactory,
+      h1: this.headingFactory('h1'),
+      h2: this.headingFactory('h2'),
+      h3: this.headingFactory('h3'),
+      h4: this.headingFactory('h4'),
+      h5: this.headingFactory('h5'),
+      h6: this.headingFactory('h6'),
+      img: this.imageFactory,
+    };
+  }
+
   async load() {
     try {
-      const { default: Document } = await this.handleImport(
-        this.props.match.params.path
+      const { params } = this.props.match;
+      const pathname = params.path || 'index';
+      const isJsonFile =
+        pathname.startsWith('reference') &&
+        (pathname.endsWith('api') || pathname.endsWith('events'));
+      const { loader, path } = readDocFile(
+        `${pathname}.${isJsonFile ? 'json' : 'md'}`
       );
+      const { default: Page } = await loader;
+
+      absolutePath = path;
+
+      // use `isMDXComponent` once https://github.com/mdx-js/mdx/pull/369 is merged
+      if (!Page.prototype) {
```

Comment:
I'm guessing this is meant to distinguish the two different possible return values of readDocFile?  Where a markdown file will return a Page instance, which has a prototype, but a `.json` file returns .. I guess, the JSON data?  So it doesn't have a prototype?  That seems pretty fragile, and certainly not reader-friendly!  Even using `.isMDXComponent` seems to violate encapsulation.  The OO solution would seem to be to wrap both data types in objects (or as components) with a common set of methods.